connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -328,8 +332,11 @@ Config targetConfig(Config sourceConfig) {
 
     AclBinding targetAclBinding(AclBinding sourceAclBinding) {
         String targetTopic = formatRemoteTopic(sourceAclBinding.pattern().name());
-        return new AclBinding(new ResourcePattern(ResourceType.TOPIC, targetTopic, PatternType.LITERAL),
-            sourceAclBinding.entry());
+        AccessControlEntry entry = sourceAclBinding.entry();
+        entry = (entry.permissionType() == AclPermissionType.ALLOW && entry.operation() == AclOperation.ALL)?
```

Comment:
Don't re-assign to a value like this. Pretend everything is "final", even if it isn't.

Don't use the ternary operator for conditional logic like this. Prefer if-statements.

I'd do something like:

```
final AccessControlEntry entry;
if (...) {
   entry = ...
} else {
   entry = ...
}
```

Reply:
You might consider extracting this as a separate method, e.g. targetAccessControlEntry().